webapp/src/components/sidebar/sidebarBoardItem.tsx

```diff
@@ -22,15 +23,25 @@ import './sidebarBoardItem.scss'
 type Props = {
     views: readonly BoardView[]
     board: Board
-    showBoard: (id?: string) => void
-    showView: (id: string, boardId?: string) => void
     activeBoardId?: string
     nextBoardId?: string
 }
 
 const SidebarBoardItem = React.memo((props: Props) => {
     const [collapsed, setCollapsed] = useState(false)
     const intl = useIntl()
+    const history = useHistory()
+    const match = useRouteMatch()
+
+    const showBoard = useCallback((boardId) => {
+        const newPath = generatePath(match.path, {...match.params, boardId: boardId || ''})
+        history.push(newPath)
+    }, [match, history])
+
+    const showView = useCallback((viewId, boardId) => {
+        const newPath = generatePath(match.path, {...match.params, boardId: boardId || '', viewId: viewId || ''})
+        history.push(newPath)
+    }, [match, history])
```

Comment:
This is kinda an special case for path generation because there are so many ways to access a board. In normal circunstances we don't need this approach to generate the route. I think the amount of repetition is still low and I think is acceptable, we can refactor it later if we see the pattern happening again and again.